File: java/src/main/java/mrk/hackerrank/HackerRank06.java
package mrk.hackerrank;

import com.google.common.base.Preconditions;

import java.util.Arrays;

//https://www.hackerrank.com/challenges/diagonal-difference/problem
public class HackerRank06 {
    public int diagonalDifference(int[][] arr) {
        Preconditions.checkArgument(
                Arrays.stream(arr).flatMapToInt(Arrays::stream).allMatch(e -> e >= -100 && e <= 100),
                "all elements must be in -100..100");
        int length = arr.length;
        int primary = 0, secondary = 0;
        for (int i = 0; i < length; i++) {
            primary += arr[i][i];
            secondary += arr[i][length - i - 1];
        }
        return Math.abs(primary - secondary);
    }
}

File: kotlin/src/main/kotlin/HackerRank06.kt
import kotlin.math.abs

fun hackerRank06(arr: Array<IntArray>): Int {
    require(arr.all { it.all { e -> e >= -100 && e <= 100 } }) { "All elements of the array must be in -100..100" }
    val length: Int = arr.size
    var primary: Int = 0
    var secondary: Int = 0
    for (i in arr.indices) {
        primary += arr[i][i]
        secondary += arr[i][length - 1 - i]
    }
    return abs(primary - secondary)
}
